src/compiler/crystal/codegen/target_machine.cr

```diff
@@ -19,7 +22,7 @@ module Crystal
           features += "+vfp2"
         end
       else
-        raise "Unsupported arch for target triple: #{target_triple}"
+        raise TargetMachine::Error.new("Unsupported architecture for target triple: #{target_triple} (only x86, x86_64, and ARM are supported)")
```

Comment:
We shouldn't list supported architectures, it's not required and could be confusing or even wrong; it's possible we added support but didn't change the message, it's possible LLVM wasn't built with a supported architectures, for example homebrew doesn't build the ARM and AArch64 targets.